cmd/server-rlimit.go

```diff
@@ -16,9 +16,33 @@
 
 package cmd
 
-import "github.com/minio/minio/pkg/sys"
+import (
+	"io/ioutil"
+	"runtime"
+	"runtime/debug"
+	"strconv"
+	"strings"
+
+	"github.com/minio/minio/pkg/sys"
+)
 
 func setMaxResources() (err error) {
+	// Set the Go runtime max threads threshold to 90% of kernel setting.
+	// Do not return any error when encountered since it is not a crucial task.
+	if runtime.GOOS == "linux" {
+		sysMaxThreadsStr, rErr := ioutil.ReadFile("/proc/sys/kernel/threads-max")
+		if rErr == nil {
+			sysMaxThreads, rErr := strconv.Atoi(strings.TrimSpace(string(sysMaxThreadsStr)))
+			if rErr == nil {
+				minioMaxThreads := (sysMaxThreads / 100) * 90
```

Comment:
Minor comment: I would multiply first and then divide (less changes of some errors) and the size of the int is large enough